pkg/odo/cli/component/delete.go

```diff
@@ -81,7 +81,33 @@ func (do *DeleteOptions) Run() (err error) {
 		}
 
 		if do.componentForceDeleteFlag || ui.Proceed(fmt.Sprintf("Are you sure you want to delete %v from %v?", do.componentName, do.Application)) {
-			err := component.Delete(do.Client, do.componentName, do.Application)
+			// Before actually deleting the component, first unlink it from any component(s) in the cluster it might be linked to
+			// We do this in three steps:
+			// 1. Get list of active components in the cluster
+			// 2. Use this list to find the components to which our component is linked and generate secret names that are linked
+			// 3. Unlink these secrets from the components
+			compoList, err := component.List(do.Client, do.Context.Application, &do.LocalConfigInfo)
+			if err != nil {
+				return err
+			}
+
+			parentComponent, err := component.GetComponent(do.Client, do.componentName, do.Context.Application, do.Context.Project)
+			if err != nil {
+				return err
+			}
+
+			componentSecrets := component.UnlinkComponents(parentComponent, compoList)
+
+			for component, secret := range componentSecrets {
+				for _, secretName := range secret {
+					err = do.Client.UnlinkSecret(secretName, component, do.Context.Application)
```

Comment:
Please add a spinner here indicating that unlinking action is happening

Reply:
Fixed